mcstatus/motd.py

```diff
@@ -0,0 +1,577 @@
+                else:
+                    tag_number = FORMATTING_TO_ANSI_TAGS[element]
+                    result += r"\e[" + tag_number + "m"
+            else:
+                foreground_color, background_color = MINECRAFT_COLOR_TO_ANSI[element]
+
+                result += r"\e[0;" + foreground_color + r"m\e[0;" + background_color + "m"
+
+        return result + r"\e[0m"
+
+
+class MotdSimplifies:
+    """Collection of MOTD simplifies.
+
+    They're just removing unused formatting, unaffected colors etc.
+    """
+
+    def __init__(self, raw: list[ParsedMotdItem]) -> None:
+        self.parsed: list[ParsedMotdItem] = raw
+
+    def get_simplifies(self) -> Iterable[Callable[[], set[int]]]:
+        """Generator with all simplifies.
+
+        They must follow this signature:
+        ``(self) -> set[int]``
+        Where ``int`` is an index of the element, which need to be removed.
+        """
+        for item in dir(self):
+            if callable(getattr(self, item)) and not item.startswith("_") and item != "get_simplifies":
+                yield getattr(self, item)
+
+    def remove_double_items(self) -> set[int]:
+        """Get indices of all doubled items that can be removed."""
+        to_remove: set[int] = set()
+
+        for index, item in enumerate(self.parsed):
+            try:
+                next_item = self.parsed[index + 1]
+            except IndexError:  # Last item (without any next item)
+                break
+
+            if item == next_item:
+                to_remove.add(index)
+
+        return to_remove
+        
+    def remove_double_colors(self) -> set[int]:
+        """Get indices of all doubled color items.
+
+        As colors (obviously) override each other, we only ever care about the last one, ignore
+        the previous ones. (for example: specifying red color, then orange, then yellow, then some text
+        will just result in yellow text)
+        """
+        to_remove: set[int] = set()
+
+        prev_color: int | None = None
+        for index, item in enumerate(self.parsed):
+            if isinstance(item, (MinecraftColor, WebColor)):
+                # If we found a color after another, remove the previous color
+                if prev_color is not None:
+                    to_remove.add(prev_color)
+                prev_color = index
+
+            # If we find a string, that's what our color we found previously applies to,
+            # set prev_color to None, marking this color as used
+            if isinstance(item, str):
+                prev_color = None
+
+        return to_remove
+
+    def remove_unused_formatting_before_color(self) -> set[int]:
+        """Remove unused formatting before color.
+
+        As colors (obviously) override formatting, we only ever care about the color, ignore
+        the formatting. (for example: specifying bold formatting, then italic, then yellow, then some text
+        will just result in yellow text)
+        """
```

Comment:
These methods no longer "remove" things, as suggested by these docstrings. Instead, they just obtain the indices.

```suggestion
        """Obtain indices of all unused formatting items before colors.

        Colors override any formatting before them, meaning we only ever care about the color, and can
        ignore all formatting before it. (For example: specifying bold formatting, then italic, then yellow, 
        will just result in yellow text.)
        """
```

Reply:
Fixed [here](https://github.com/py-mine/mcstatus/compare/ab68edac24159f7185f462310971171426167a68..676680ad8a2d0f97a9421da968add8149544a19c).